packages/core/src/root.ts

```diff
@@ -215,6 +227,40 @@ export function createRemoteRoot<
 
       return text;
     },
+    createFragment() {
+      const id = `${currentId++}`;
+
+      const internals: FragmentInternals = {
+        children: strict ? Object.freeze([]) : [],
+      };
+
+      const fragment: RemoteFragment<Root> = {
+        kind: KIND_FRAGMENT,
+        get children() {
+          return internals.children;
+        },
+        appendChild: (child) =>
+          appendChild(
+            fragment,
+            normalizeChild(child, remoteRoot),
+            internals,
+            rootInternals,
+          ),
+        removeChild: (child) =>
+          removeChild(fragment, child, internals, rootInternals),
+        insertChildBefore: (child, before) =>
+          insertChildBefore(fragment, child, before, internals, rootInternals),
+
+        // Just satisfying the type definition, since we need to write
+        // some properties manually.
+        ...EMPTY_OBJECT,
+      };
+
+      makePartOfTree(fragment, rootInternals);
+      makeRemote(fragment, id, remoteRoot);
+
+      return fragment;
+    },
     appendChild: (child) =>
       appendChild(
```

Comment:
Should we allow calling root.appendChild and component.appendChild with a fragment? It would make vanilla more powerful. You can create a fragment and do dynamic operations on it and then be able to append it to any node which allows flexibility in how the component tree is built.

Reply:
I think it makes sense to allow fragment being as a child. 

cc @lemonmade

Reply:
We will do a follow up PR for this.